src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +1003,64 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn lockfile_path(
+    lockfile_path: Option<&Path>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<PathBuf>> {
+    let path;
+
+    if let Some(lockfile_path) = lockfile_path {
+        if !gctx.cli_unstable().unstable_options {
+            bail!("`--lockfile-path` option requires `-Zunstable-options`")
+        }
+
+        if lockfile_path.is_absolute() {
+            path = lockfile_path.to_path_buf();
+        } else {
+            path = gctx.cwd().join(lockfile_path);
+        }
```

Comment:
This should be sufficient.

* [`Path::join`](https://doc.rust-lang.org/1.80.0/std/path/struct.Path.html#method.join)  replaces the original path if the given is absolute.
* Place the variable declaration together when binding.


```suggestion
        let path = gctx.cwd().join(lockfile_path);
```

Reply:
Good catch, thanks!